### PR TITLE
chore(*): add pre-commit hooks for formatting

### PR DESCRIPTION
**Pull request by RaitoBezarius**
This adds pre-commit hooks for formatting, because I usually just trust whatever people prefer locally :).
We could add, e.g. rustfmt, but I prefer @sonmarcho that you tell me what you prefer regarding this. Pre-commit hooks can be pesky and annoying.

Signed-off-by: Ryan Lahfa <ryan.lahfa@inria.fr>

### CODE REVIEW

**Review by Nadrieril — APPROVED** [review]
We recently added CI checks for formatting, this makes a lot of sense. Could you do the same (with rustfmt also) on Charon's side?

**Comment by RaitoBezarius** [comment]
> We recently added CI checks for formatting, this makes a lot of sense. Could you do the same (with rustfmt also) on Charon's side?

Yes, of course!

**Comment by sonmarcho** [comment]
In practice what does this do? It only uses pre commit hooks if you're using nix, right? If that's the case and you consider it's idiomatic when using nix, I'm fine.

File: flake.nix [hunk 1]
```diff
@@ -183,9 +189,26 @@
...
-                  aeneas-verify-fstar
-                  aeneas-verify-coq
-                  aeneas-verify-hol4
-                  aeneas-check-tidiness; };
+            aeneas-verify-fstar
+            aeneas-verify-coq
+            aeneas-verify-hol4
+            aeneas-check-tidiness;
+
+          pre-commit-check = pre-commit-hooks.lib.${system}.run {
```
Comment (Nadrieril) [inline flake.nix:197]: I'm not a huge fan of pre-commit hooks because I rebase a lot... I does make sense though, we can try it for a bit

Comment (sonmarcho) [reply flake.nix:197]: Can you add a comment here to explain what this does (this formats the code at commit time, etc.)?

**Comment by Nadrieril** [comment]
Now that my editor formats everything automatically I'm not longer keen on having this

**Comment by sonmarcho** [comment]
This PR seems dead, so I'm closing it. Feel free to reopen it if you really want this feature.

**Pull request closed without merging**